Smartpad-macOS/Smartpad-macOS/Logic/ConnectionManager.swift

```diff
@@ -59,23 +61,22 @@ extension ConnectionManager{
     func session(_ session: MCSession, peer peerID: MCPeerID, didChange state: MCSessionState) {
         switch state {
             case .connected:
-                print("Connected: \(peerID.displayName)")
+//                print("Connected: \(peerID.displayName)")
                 DispatchQueue.main.async {
-                    self.pairVC?.setPairLabel(label: "Connected: \(peerID.displayName)")
-                    self.pairVC?.updateConnectionView(status: ConnStatus.PairedAndConnected)
+                    self.mainVC?.updateConnStatus(status: ConnStatus.PairedAndConnected, peerName: peerID.displayName)
                 }
             
             case .connecting:
-                print("Connecting: \(peerID.displayName)")
+                // Just stay in the same state, UI will be hidden while we are connected anyways.
+//                print("Connecting: \(peerID.displayName)")
                 DispatchQueue.main.async {
```

Comment:
I guess you could remove the main Dispatch queue as well.

Reply:
Makes sense, moved the async dispatch to MainViewController (where it should have been in the first place)